src/main.cc

```diff
@@ -86,6 +87,28 @@ class Spellchecker : public Nan::ObjectWrap {
     }
   }
 
+  static NAN_METHOD(CheckSpellingCallback) {
+    Nan::HandleScope scope;
+    if (info.Length() < 2) {
+      return Nan::ThrowError("Bad argument");
+    }
+
+    Handle<String> string = Handle<String>::Cast(info[0]);
+    if (!string->IsString()) {
+      return Nan::ThrowError("Bad argument");
+    }
+
+    Nan::Callback *callback = new Nan::Callback(info[1].As<v8::Function>());
```

Comment:
In other places, I've dealt with this by *hiding* the callback-based method so that you can guarantee it'll never be called with invalid arguments. Something like:

```js
// This is the native function
const {checkSpellingAsync} = Spellchecker.prototype

// Override the native callback-based method with a promise-based one.
// This way, it's not possible to call the native one with invalid arguments.
Spellchecker.prototype.checkSpellingAsync = function (corpus) {
  return new Promise(function (resolve, reject) {
   checkSpellingAsync.call(this, corpus, (err, result) => {
     // ...
   }
}

```